Add list users snippet, concept and testing (#732)
openfga/openfga.dev: docs/content/concepts.mdx

  expectedResults={['document:otherdoc', 'document:planning']}
/>

For more information, see the [Relationship Queries page](./interacting/relationship-queries.mdx) and the official [List Objects API Reference](/api/service#Relationship%20Queries/ListObjects).

</details>
<details>
<summary>

## What Is A List Users Request?

A **list users request** is a call to the <ProductName format={ProductNameFormat.LongForm}/> list users endpoint that returns all users of a given type that have a specified relationship with an object.

</summary>

> List users requests are completed using the `listUsers` methods in the <ProductName format={ProductNameFormat.ShortForm}/> SDKs ([JavaScript SDK](https://www.npmjs.com/package/@openfga/sdk)/[Go SDK](https://github.com/openfga/go-sdk)/[.NET SDK](https://www.nuget.org/packages/OpenFga.Sdk)) by manually calling the [list users endpoint](/api/service#Relationship%20Queries/ListUsers) using curl or in your code.

Author: ```suggestion
List users requests are completed using the relevant `ListUsers` method in SDKs, the `fga query list-users` command in the CLI, or by manually calling the [ListUsers endpoint](/api/service#Relationship%20Queries/ListUsers) using curl or in your code.
```

> List users requests are completed using the `listUsers` methods in the <ProductName format={ProductNameFormat.ShortForm}/> SDKs ([JavaScript SDK](https://www.npmjs.com/package/@openfga/sdk)/[Go SDK](https://github.com/openfga/go-sdk)/[.NET SDK](https://www.nuget.org/packages/OpenFga.Sdk)) by manually calling the [list users endpoint](/api/service#Relationship%20Queries/ListUsers) using curl or in your code.

Author: Added a suggestion that reduces to "SDKs" and adds CLI"

> List object requests are completed using the `listUsers` methods in the <ProductName format={ProductNameFormat.ShortForm}/> SDKs ([JavaScript SDK](https://www.npmjs.com/package/@openfga/sdk)/[Go SDK](https://github.com/openfga/go-sdk)/[.NET SDK](https://www.nuget.org/packages/OpenFga.Sdk)) by manually calling the [list users endpoint](/api/service#Relationship%20Queries/ListUsers) using curl or in your code.

Reviewer: ```suggestion
List users requests are completed using the `listUsers` methods in the <ProductName format={ProductNameFormat.ShortForm}/> SDKs ([JavaScript SDK](https://www.npmjs.com/package/@openfga/sdk)/[Go SDK](https://github.com/openfga/go-sdk)/[.NET SDK](https://www.nuget.org/packages/OpenFga.Sdk)) by manually calling the [list users endpoint](/api/service#Relationship%20Queries/ListUsers) using curl or in your code.
```

> The list users endpoint responds with a list of users and excluded users for a given type that have the specificed relationship with an Object.

Reviewer: ```suggestion
The list users endpoint responds with a list of users and excluded users for a given type that have the specificed relationship with an object.
```
